apps/small_filters/conv3x3a16.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Halide.h"
+
+using namespace Halide;
+using namespace Halide::Internal;
+IRPrinter irp(std::cerr);
+
+class Conv3x3a16 : public Generator<Conv3x3a16> {
+public:
+    // Takes an 8 bit image; one channel.
+    Input<Buffer<uint8_t>> input{"input", 2};
+    Input<Buffer<int8_t>> mask{"mask", 2};
+
+    // Outputs an 8 bit image; one channel.
+    Output<Buffer<uint8_t>> output{"output", 2};
+    Var x{"x"}, y{"y"};
+    void generate() {
+        Func bounded_input{"input_bounded"};
+
+        bounded_input(x, y) = BoundaryConditions::repeat_edge(input)(x, y);
+
+        Expr sum = cast<int16_t>(0);
+        for (int i = -1; i <= 1; i++) {
+            for (int j = -1; j <= 1; j++) {
+                sum += cast<int16_t>(bounded_input(x+j, y+i)) * cast<int16_t>(mask(j+1, i+1));
+            }
+        }
+        output(x, y) = cast<uint8_t>(clamp(sum >> 4, 0, 255));
+        bounded_input.compute_root();
+    }
+
+    void schedule() {
+        Var xi{"xi"}, yi{"yi"};
+
+        auto set_min_0 = [this](int dim) {
```

Comment:
I think this lambda should be removed and the code rewritten to just call it directly. Note that there's only actually a single unique line for each dimension:

    output_buffer.dim(i).set_min(0);

The other two lines are only needed once (and can remove the call to output_buffer() in the schedule for HVX below).